app/connectors/HmrcTierConnectorWrapped.scala

```diff
@@ -20,35 +20,57 @@ import play.api.http.Status
 import play.api.libs.json.{JsValue, Json}
 import play.api.{Configuration, Logging}
 import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
-import uk.gov.hmrc.http.HttpReads.Implicits
-
+import java.util.UUID.randomUUID
 import javax.inject.Inject
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
-
+import uk.gov.hmrc.http.HttpReads.Implicits.readRaw
 class HmrcTierConnectorWrapped @Inject()(val http: HttpClient, configuration: Configuration) extends Logging {
 
   val serviceOriginatorIdKey: String = configuration.get[String]("microservice.services.nps.originatoridkey")
   val serviceOriginatorId: String = configuration.get[String]("microservice.services.nps.originatoridvalue")
+  val CORRELATION_HEADER = "CorrelationId"
+
+  private def buildHeaders(correlationId: String): Seq[(String, String)] =
+    Seq(
+      serviceOriginatorIdKey -> serviceOriginatorId,
+      CORRELATION_HEADER     -> correlationId
+    )
 
-  private val extraHeaders: Seq[(String, String)] = Seq(serviceOriginatorIdKey -> serviceOriginatorId)
+  def generateNewUUID: String = randomUUID.toString
+
+  private[connectors] def getCorrelationId(hc: HeaderCarrier): String = {
+    val CorrelationIdPattern = """.*([A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}).*""".r
```

Comment:
This is actually the requestId pattern. It could also be placed outside of the method as it will never change.